### ISSUE TITLE
Enhance Test-DbaBuild to include all Components of SQL Server

### ISSUE THREAD

**Issue by lostsole** [sqlcollaborative/dbatools#7135]
### Summary of new feature

Enhance Test-DbaBuild to include all Components of SQL Server, SSAS, SSIS and SSRS.

### Proposed technical details (if applicable)

From this URL:https://www.mssqltips.com/sqlservertip/3549/inventory-sql-server-services-version-and-edition/
Something like this that returns all the inventory and build of all SQL components.

### Latest version of dbatools as of writing

1.0.137

**Comment by potatoqualitee** [comment]
I don't think we track individual components, unless their version matches up with the SQL Server version. If so, this would be doable and interesting. What you think, @jpomfret ?

**Comment by potatoqualitee** [comment]
I researched this against the current `Test-DbaBuild` implementation and Microsoft's component servicing model. A single “include every component” switch is not currently a bounded extension because the components do not share one build namespace or one discovery transport.

### Current dbatools boundary

`Test-DbaBuild` consumes `Get-DbaBuild` results and compares Database Engine versions against the dbatools build-reference index. `-SqlInstance` is an Engine/SMO connection, so the command currently has neither discovery nor reference data for SSAS, SSIS, or SSRS.

### Component differences

- **SSAS:** beginning with SQL Server 2017, Analysis Services and Database Engine build numbers no longer match even though they can be serviced by the same CU package. Microsoft says the build systems are separate and recommends checking `msmdsrv.exe` component file versions: https://learn.microsoft.com/en-us/analysis-services/instances/analysis-services-component-version?view=sql-analysis-services-2025
- **SSIS:** its installed version generally aligns with the SQL Server version, but discovering the actually installed/serviced component is a Windows installation/file/service inventory task, not an Engine query. Microsoft's component-version guidance treats it separately: https://learn.microsoft.com/en-us/troubleshoot/sql/releases/components-client-tools-versions
- **SSRS:** since SQL Server 2017 it is a separate download with its own product/file versions and release stream. The SSRS release notes show versions that do not match Database Engine CU numbers: https://learn.microsoft.com/en-us/sql/reporting-services/release-notes-reporting-services?view=sql-server-ver17. Starting with SQL Server 2025, on-premises reporting is consolidated under Power BI Report Server, adding another product boundary: https://learn.microsoft.com/en-us/sql/reporting-services/what-s-new-in-sql-server-reporting-services-ssrs?view=sql-server-ver17

Discovery also differs: Engine via TDS/SMO, SSAS via AMO or `msmdsrv.exe`, SSIS via Windows installation/service/file metadata, and SSRS via its web endpoint/configuration/file metadata. Remote file/service discovery introduces Windows remoting, firewall, permissions, and non-Windows limitations.

### Recommended decomposition

1. **Inventory first:** add or extend a component-inventory command that reports installed component, instance/name, product version, file version, discovery source, and host. `Get-DbaService` is useful input but currently does not provide the authoritative serviced build for every component.
2. **Reference data second:** define separate, source-backed indexes per component. Reusing the Engine build index for SSAS/SSRS would produce false compliance results.
3. **Compliance last:** extend `Test-DbaBuild` with an explicit `-Component` contract only after discovery and reference sources exist. Output needs `Component` and `ReferenceSource` so unlike version schemes are never compared silently.

A sensible first PR would be **SSAS-only** inventory/compliance because Microsoft publishes an explicit component-file mapping in CU documentation. SSIS could follow after defining its authoritative file set. SSRS should likely remain separate because its post-2017 servicing and 2025 product transition are independent.

Before implementation, the issue needs decisions on:

- whether the command may require Windows remoting/file access;
- which components and versions are in the first scope;
- who maintains the additional build indexes and from which authoritative sources;
- whether unreachable components are noncompliant or indeterminate.

My recommendation is not to add a broad switch to `Test-DbaBuild` yet. Approve a small SSAS inventory/reference-data design first, then expand component by component.

**Comment by andreasjordan** [comment]
Chrissy's analysis from July explains why an "all components" switch has no bounded implementation: SSAS, SSIS and SSRS share neither a build namespace nor a discovery transport with the Database Engine, and the build reference that `Test-DbaBuild` compares against covers the Engine only. Nothing has changed since, and nobody has picked the item up in the five years it has been open.

I suggest closing this as not planned. What the linked mssqltips script inventories is available today outside `Test-DbaBuild`: the file versions of the component executables (`msmdsrv.exe`, `ISServerExec.exe`, `ReportingServicesService.exe`), which is also what Microsoft points to for servicing them. If a bounded slice comes up later, for example SSRS through its REST API, it can return as its own request.

created by Claude and reviewed by Andreas Jordan
